diff --git a/src/main.c b/src/main.c
index ca4797d..6766f36 100644
--- a/src/main.c
+++ b/src/main.c
@@ -169,13 +169,13 @@ void *run_log_thread(void *args) {
     end_pos = lseek(fd,0L,SEEK_END);
     offset = end_pos - last_read_pos;
     if (offset > 0) {
-      char *message = malloc(offset + 1);
+      char message[2048];
       lseek(fd, last_read_pos, SEEK_SET);
-      bytesread = read(fd, (void*)message, offset);
+      bytesread = read(fd, (void*)&message, offset);
       message[bytesread+1] = '\0';
       last_read_pos += bytesread;
-      leftover = ui_display_log_chunk(ui, message, leftover);
-      free(message);
+      leftover = ui_display_log_chunk(ui, (char*)&message, leftover);
+     
     }
     nanosleep(&interval, NULL);
   }
@@ -232,7 +232,7 @@ void* gui_loop(void*args) {
       int b = 0;
       
       do {
-        message = get_message(ui);
+        message = get_user_input(ui);
         if(strcmp(message,":close") == 0) {
           b = 1;
           break;
